yt_xarray/accessor/accessor.py

```diff
@@ -15,6 +15,42 @@ def __init__(self, xarray_obj):
         self._bbox = {}
         self._field_grids = defaultdict(lambda: None)
 
+    def load_grid_from_callable(self):
+        def _read_data(handle):
+            def _reader(grid, field_name):
+                ftype, fname = field_name
+                si = grid.get_global_startindex()
+                ei = si + grid.ActiveDimensions
+                var = getattr(
+                    handle, fname
+                )  # variables are top-level attrs of xr datasets
+                data = var[si[0] : ei[0], si[1] : ei[1], si[2] : ei[2]]
+                return data.values
+
+            return _reader
+
+        reader = _read_data(self._obj)
+
+        fields = list(self._obj.data_vars)
+        shape = self._obj.data_vars[fields[0]].shape
+
+        # again, need to account for stretched grid here...
+        data = {_: reader for _ in fields}
+        data.update(
+            {
+                "left_edge": [0.0, 0.0, 0.0],
```

Comment:
Is `left_edge` always 0?  and `right_edge` always 1?

Reply:
nope! that's one of the things to fix :)

Reply:
appropriate values are now being pulled from the bounding box!